Automattic/vip-go-mu-plugins
# Introduce Attendant class for Jetpack user management

## Description

 Introduces new `Attendant` class that will be responsible for handling the jetpack connection user.

By default, sites will continue using the same `wpcomvip` user for connections. Upon enabling via the `vip_jetpack_connection_pilot_use_org_owner` filter, the new customer-specific user will be used as the connection owner. Once the filter is enabled, existing connections will fail the connection test with the `jp-cxn-pilot-has-legacy-vip-owner` error which will trigger a (re)connection attempt that will make the new user the owner. But until the filter is enabled, all will stay the same.

Additionally using this Attendant class to lock down the bot user with additional security restrictions. In the future, after all sites have switched over, we'll be able to phase out all the various `wpcomvip`-specific controls throughout the codebase and can even work towards removing that user from sites as it is no longer used for anything else.

## Changelog Description

### Plugin Updated: VIP Jetpack

Introduce new `Attendant` class for managing the Jetpack connection's primary user.

## Deploy Notes

There is a small change that needs to take place in the `Provisioner` class as well that must go out alongside this change.

## Steps to Test

Default User:
1) Create a new subsite
2) Run `wp jetpack-start connect --url=example.com/test1`
3) Ensure JP is connected underneath the `wpcomvip` user.

New User:
1) Add `add_filter( 'vip_jetpack_connection_pilot_use_org_owner', '__return_true' )` to the site (before plugins loaded priority 25)
2) Create another subsite
3) Run `wp jetpack-start connect --url=example.com/test2`
4) Ensure JP is connected underneath the new user.

Migration:
1) On the first subsite, ensure the filter is now present.
2) Disconnect JP: `wp jetpack disconnect blog --url=example.com/test1`
3) Re-connect JP: `wp jetpack-start connect --url=example.com/test1`
4) Ensure the new user is now the connection owner.
